pandas/io/parsers.py

```diff
@@ -1183,13 +1193,22 @@ def __init__(self, src, **kwds):
         self._implicit_index = self._reader.leading_cols > 0
 
     def _set_noconvert_columns(self):
-        names = self.names
+        # By the time this function is called,
+        # 'names' and 'orig_names' will either
+        # be identical, or 'names' will be a
+        # strict subset of 'orig_names'
+        identical = (len(self.names) == len(self.orig_names))
```

Comment:
They are not identical when you specify `usecols`.  If you see where this method is called, it is called right after `self.names` is filtered by `usecols`.  The fix is I set `self.orig_names` to `self.names` before filtering so that we have a copy of the original columns before filtering.  The fact we didn't have a copy of the unfiltered column list was why this bug emerged.